python/src/main/python/drivers/vkrun.py

```diff
@@ -139,20 +139,26 @@ def run_android(vert, frag, json, skip_render):
 
     # Busy wait
     deadline = time.time() + TIMEOUT_APP
-    crash = False
-    done = False
+
+    status = 'UNEXPECTED_ERROR'
+
+    # If the app never starts, status remains as UNEXPECTED_ERROR.
+    # Once the app starts, status becomes TIMEOUT.
+    # If we break out of the loop below, the status is updated just before.
 
     while time.time() < deadline:
 
-        # Begin the busy-wait loop by sleeping to let the app start
-        # properly. The 'adb shell am start' may return before the app is
-        # actually started (this is all asynchronous), in which case we may not
-        # be able to detect the app pid.
-        time.sleep(0.1)
+        # Don't pass here until app has started.
+        if status == 'UNEXPECTED_ERROR':
+            if adb('shell test -f ' + ANDROID_SDCARD + '/STARTED').returncode != 0:
+                continue
```

Comment:
I suggest to add a time.sleep(0.1) while waiting for the STARTED file, otherwise it is busy-looping without any sleep.

Reply:
OK done.